src/lib/chat/matrix-client.ts

```diff
@@ -118,10 +119,13 @@ export class MatrixClient implements IChatClient {
       await room.decryptAllEvents();
       await room.loadMembersIfNeeded();
       const membership = room.getMyMembership();
-      if (membership === MembershipStateType.Invite) {
-        if (!(await this.autoJoinRoom(room.roomId))) {
-          failedToJoin.push(room.roomId);
-        }
+      if (
```

Comment:
1. What happens if `getConversations` is called again? Do we end up doubling up on the event handlers?
2. We should also do this for Channels, I think. This is not a conversation specific thing.
3. This `if` statement is a weird nested combination of logic and functionality. I think we should probably tidy that up.

Reply:
Hm weird I didn't get an email notification for the review!

re point 2. Yeah I think it makes sense to add to Channels too.
re point 3. Weird, not sure why I added the initialize in an if statement. Reverted and added just in the loop.

Reply:
re point 1. How about adding/using the off method before the on method in the `initializeRoomEventHandlers` to safe guard this? We'd ensure that each room will only have one instance of the event handler attached, regardless of how many times getConversations is called. I've pushed this proposed change, let me know what you think @dalefukami